### ISSUE TITLE
Not working with xml

### ISSUE THREAD

**Issue by narendrakothamire** [JoanZapata/android-iconify#137]
Added dependency 

compile 'com.joanzapata.iconify:android-iconify-material:2.1.0'
compile 'com.joanzapata.iconify:android-iconify-fontawesome:2.1.0' 

Added code in Application class

Iconify.with(new MaterialModule())
                .with(new FontAwesomeModule());

<com.joanzapata.iconify.widget.IconTextView
        android:layout_toRightOf="@+id/question_textView"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:textColor="#fff"
        android:text="{fa-heart-o}" />

I used this but i get ouptut as {fa-hear-o}

**Comment by sytolk** [comment]
Xml not work for me too. I have success only with programmatically set icon.

I have use only material icons.This is mu maven project dependency
```xml
<dependency>
            <groupId>com.joanzapata.iconify</groupId>
            <artifactId>android-iconify</artifactId>
            <type>aar</type>
            <version>2.1.0</version>
            <exclusions>
                <exclusion>
                    <groupId>com.android.support</groupId>
                    <artifactId>support-v4</artifactId>
                </exclusion>
            </exclusions>
        </dependency>

        <dependency>
            <groupId>com.joanzapata.iconify</groupId>
            <artifactId>android-iconify-material</artifactId>
            <type>aar</type>
            <version>2.1.0</version>
            <exclusions>
                <exclusion>
                    <groupId>com.joanzapata.iconify</groupId>
                    <artifactId>android-iconify</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
```
In Application.java
```java
Iconify.with(new MaterialModule());
```
layout.xml file:
```xml
<com.joanzapata.iconify.widget.IconTextView
            android:id="@+id/bluetoothButton"
            android:layout_width="0dp"
            android:layout_height="wrap_content"
            android:layout_weight=".2"
            android:onClick="onBluetoothButton"
            android:text="{md_bluetooth @dimen/main_button}"
            android:shadowColor="#22000000"
            android:shadowDx="3"
            android:shadowDy="3"
            android:shadowRadius="1"
            android:textColor="#FFFFFFFF"/>
```

**Comment by JoanZapata** [comment]
@sytolk `md_bluetooth` is invalid using v2, you're supposed to use `md-bluetooth`.
@narendrakothamire Are you using an `IconTextView` ?

**Comment by sytolk** [comment]
I think that its "_" 
https://github.com/JoanZapata/android-iconify/blob/master/android-iconify-material/src/main/java/com/joanzapata/iconify/fonts/MaterialIcons.java#L75

I have copy in xml the same property like programmatically set:
```java
new DrawerItem(getString(R.string.drawer_entry_home), MaterialIcons.md_home..
```
Or you replace "_" with "-" in library?

**Comment by JoanZapata** [comment]
Read the file to the [bottom](https://github.com/JoanZapata/android-iconify/blob/master/android-iconify-material/src/main/java/com/joanzapata/iconify/fonts/MaterialIcons.java#L810).

The `-` version is what keys look like in every icon font when you use them on the web and I wanted Iconify to work the same, but Java doesn't allow `-` in names, so I had to use `_` instead for the constants.

**Comment by sytolk** [comment]
XML works for me with 2.1.0 version thanks @JoanZapata maybe Readme can be updated with this.

The - version is what keys look like in every icon font when you use them on the web and I wanted Iconify to work the same, but Java doesn't allow - in names, so I had to use _ instead for the constants.

**Comment by JoanZapata** [comment]
I don't think that's needed. The `README` uses `-` already, the demo app too. 
You're not supposed to use `_` at all.

**Comment by sytolk** [comment]
Yes I see that in README example have "-" instead of "_" but I think that programmers will believe what they see in the source code ;)

**Comment by narendrakothamire** [comment]
@JoanZapata  Yes i am using IconTextView

**Comment by Plastix** [comment]
XML Isn't working for me either.

**Comment by JoanZapata** [comment]
@narendrakothamire Is you `Application` subclass declared in your `AndroidManifest.xml`?

**Comment by christiansoe** [comment]
I can't show icon in xml too.

**Comment by miladna7** [comment]
please notice that, you should write 
Iconify.with(new ....

before  the   
setContentView(R.....);

it is very important.

**Comment by jhonoryza** [comment]
I can't show icon in xml too, is it a bug?

**Comment by miladna7** [comment]
No, i think you, New it after the
setcontentview
.
U should new it before setting the layout for your activity.

**Comment by jhonoryza** [comment]
ok my bad, now i set new before setcontentview finally i can see the icon after i built and run the code, but there is no live preview in xml, is it alright?

**Comment by miladna7** [comment]
Oh, your welcoem dear.
It was my problem for  couple of days, too :))).
Yep, there isn't any live view.

Now, they can close this.
